account_invoice_merge/tests/test_account_invoice_merge.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo.tests.common import TransactionCase
+from odoo.exceptions import UserError
+
+
+class TestAccountInvoiceMerge(TransactionCase):
+    """
+        Tests for Account Invoice Merge.
+    """
+    def setUp(self):
+        super(TestAccountInvoiceMerge, self).setUp()
+        self.par_model = self.env['res.partner']
+        self.context = self.env['res.users'].context_get()
+        self.acc_model = self.env['account.account']
+        self.inv_model = self.env['account.invoice']
+        self.inv_line_model = self.env['account.invoice.line']
+        self.wiz = self.env['invoice.merge']
+
+        self.partner1 = self._create_partner()
+        self.partner2 = self._create_partner()
+
+        self.invoice_account = self.acc_model.search(
+            [('user_type_id',
+              '=',
+              self.env.ref('account.data_account_type_receivable').id
+              )], limit=1)
+
+        self.invoice_line1 = self._create_inv_line(self.invoice_account.id)
+        self.invoice_line2 = self.invoice_line1.copy()
+        self.invoice_line3 = self._create_inv_line(self.invoice_account.id)
+
+        self.invoice1 = self._create_invoice(
+            self.partner1, 'A', self.invoice_line1)
+        self.invoice2 = self._create_invoice(
+            self.partner1, 'B', self.invoice_line2)
+        self.invoice3 = self._create_invoice(
+            self.partner2, 'C', self.invoice_line3)
+
+    def _create_partner(self):
+        partner = self.par_model.create({
+            'name': 'Test Partner',
+            'supplier': True,
+            'company_type': 'company',
+        })
+        return partner
+
+    def _create_inv_line(self, account_id):
+        inv_line = self.inv_line_model.create({
+            'name': 'test invoice line',
+            'account_id': account_id,
+            'quantity': 1.0,
+            'price_unit': 3.0,
+            'product_id': self.env.ref('product.product_product_8').id
```

Comment:
direct use  `self.ref('product.product_product_8')` no need to `.id`

Reply:
Hum, I'm not fan to pass the record himself to create or edit a record. For me, product_id should be the ID and not record.